mteb/abstasks/AbsTaskMultilabelClassification.py

```diff
@@ -0,0 +1,146 @@
+
+    def _add_main_score(self, scores):
+        if self.metadata_dict["main_score"] in scores:
+            scores["main_score"] = scores[self.metadata_dict["main_score"]]
+        else:
+            logger.warn(
+                f"main score {self.metadata_dict['main_score']} not found in scores {scores.keys()}"
+            )
+
+    def evaluate(self, model, eval_split="test", train_split="train", **kwargs):
+        if not self.data_loaded:
+            self.load_data()
+
+        if self.is_multilingual:
+            scores = {}
+            for lang in self.dataset:
+                logger.info(
+                    f"\nTask: {self.metadata_dict['name']}, split: {eval_split}, language: {lang}. Running..."
+                )
+                scores[lang] = self._evaluate_monolingual(
+                    model, self.dataset[lang], eval_split, train_split, **kwargs
+                )
+                self._add_main_score(scores[lang])
+        else:
+            logger.info(
+                f"\nTask: {self.metadata_dict['name']}, split: {eval_split}. Running..."
+            )
+            scores = self._evaluate_monolingual(
+                model, self.dataset, eval_split, train_split, **kwargs
+            )
+            self._add_main_score(scores)
+
+        return scores
+
+    def _evaluate_monolingual(
+        self, model, dataset, eval_split="test", train_split="train", **kwargs
+    ):
+        train_split = dataset[train_split]
+        eval_split = dataset[eval_split]
+        params = {"k": self.k, "batch_size": self.batch_size}
+        params.update(kwargs)
+
+        scores = []
+        test_cache, idxs = (
+            None,
+            None,
+        )  # we store idxs to make the shuffling reproducible
+        for i in range(self.n_experiments):
+            logger.info(
+                "=" * 10 + f" Experiment {i+1}/{self.n_experiments} " + "=" * 10
+            )
+            # Bootstrap `self.samples_per_label` samples per label for each split
+            X_sampled, y_sampled, idxs = self._undersample_data(
+                train_split["text"], train_split["label"], self.samples_per_label, idxs
+            )
+            binarizer = MultiLabelBinarizer()
+            y_train = binarizer.fit_transform(y_sampled)
+            y_test = binarizer.transform(eval_split["label"])
+            evaluator = kNNMultiLabelClassificationEvaluator(
+                X_sampled,
+                y_train,
+                eval_split["text"],
+                y_test,
+                **params,
+            )
+            scores_exp, test_cache = evaluator(model, test_cache=test_cache)
+            scores.append(scores_exp)
+
+        if self.n_experiments == 1:
+            return scores[0]
+        else:
+            avg_scores = {k: np.mean([s[k] for s in scores]) for k in scores[0].keys()}
+            std_errors = {
+                k + "_stderr": np.std([s[k] for s in scores]) for k in scores[0].keys()
+            }
```

Comment:
I would love to also add the raw scores (e.g. to allow calculating win rates, CI etc.)

Reply:
Again, is this something we should address in this PR or in another one with multiple tasks at once?